### PR TITLE
Misc Patches for FHIR Converter

### PR DESCRIPTION
**Pull request by jonahkaye**
Refs: #[1449](https://github.com/metriport/metriport-internal/issues/1449)

### Description

- patch to observation for not having string handling
- pre-emptively patch string handling everywhere else. values that are strings that are not wrapped in strings cause malformed json errors.
- patch for observation having weird chars in value._ that cause malformed json
- routeCode patch

### Testing

_[Regular PRs:]_

- Local
  - [x] local fhir mr gen
  - [x] test on a dozen patients 

### Release Plan

- [x] Lets bundle up some more patches
- [x] this points to master
- [ ] Merge this

### CODE REVIEW

File: packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js [hunk 1]
```diff
@@ -1017,4 +1017,14 @@ module.exports.external = [
...
     },
   },
+  {
+    name: "parseReferenceData",
+    description: "Escapes new line and other special chars when parsing ._ fields",
+    func: function (referenceData) {
+      if (referenceData === undefined) {
+        return "";
+      }
+      return JSON.stringify(referenceData).slice(1, -1);
```
Comment (leite08) [inline packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js:1027]: This doesn't seem to match the description/name of the function.

File: packages/fhir-converter/src/templates/cda/Resources/MedicationAdministration.hbs [hunk 2]
```diff
@@ -20,12 +20,12 @@
                 "route":{{>DataType/CodeableConcept.hbs code=medicationAdministration.routeCode}}
                 "dose":
                 {
-                    "value":{{medicationAdministration.doseQuantity.value}},
+                    "value":"{{medicationAdministration.doseQuantity.value}}",
```
Comment (leite08) [inline packages/fhir-converter/src/templates/cda/Resources/MedicationAdministration.hbs:23]: Are we sure to move all these numbers to string? Confirming b/c there's no context for this on the PR description or specific notes

File: packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js [hunk 3]
```diff
@@ -1017,4 +1017,14 @@ module.exports.external = [
       return str.startsWith(substr);
     },
   },
+  {
+    name: "parseReferenceData",
+    description: "Escapes new line and other special chars when parsing ._ fields",
+    func: function (referenceData) {
+      if (referenceData === undefined) {
```
Comment (leite08) [inline packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js:1024]: prefer `==` over `===` for `undefined`

File: packages/fhir-converter/src/templates/cda/Resources/MedicationAdministration.hbs (hunk 2 above)
Comment (jonahkaye) [reply packages/fhir-converter/src/templates/cda/Resources/MedicationAdministration.hbs:23]: yah sorry about that. [Context](https://metriport.slack.com/archives/C04T256DQPQ/p1705998742866599?thread_ts=1705997203.233089&cid=C04T256DQPQ)

File: packages/fhir-converter/src/templates/cda/Resources/MedicationStatement.hbs [hunk 4]
```diff
@@ -25,7 +25,7 @@
                 {
                     "doseQuantity":
                     {
-                        "value":{{medicationStatement.doseQuantity.value}},
+                        "value":"{{medicationStatement.doseQuantity.value}}",
```
Comment (jonahkaye) [inline packages/fhir-converter/src/templates/cda/Resources/MedicationStatement.hbs:28]: fyi Raf this patch would fix your error here:
https://metriport.slack.com/archives/C06CQC7GQG3/p1706031381120459?thread_ts=1706027906.239089&cid=C06CQC7GQG3

File: packages/fhir-converter/src/templates/cda/Resources/Observation.hbs [hunk 5]
```diff
@@ -68,11 +68,11 @@
         {{else if observationEntry.value.value}}
             "valueQuantity":
             {
-                "value":{{observationEntry.value.value}},
+                "value":"{{observationEntry.value.value}}",
                 "unit":"{{observationEntry.value.unit}}",
             }, 
         {{else}}
-            "valueString":"{{observationEntry.value._}}",
+            "valueString":"{{{parseReferenceData observationEntry.value._}}}",
```
Comment (jonahkaye) [inline packages/fhir-converter/src/templates/cda/Resources/Observation.hbs:75]: this patches this error: https://metriport.slack.com/archives/C06CQC7GQG3/p1706031364826159?thread_ts=1706027906.239089&cid=C06CQC7GQG3

File: packages/fhir-converter/src/templates/cda/Resources/MedicationStatement.hbs [hunk 6]
```diff
@@ -20,12 +20,12 @@
         [
             {
                 "text":"{{medicationStatement.text._}}",
-                "routeCode":{{>DataType/CodeableConcept.hbs code=medicationStatement.routeCode}},
+                "route":{{>DataType/CodeableConcept.hbs code=medicationStatement.routeCode}},
```
Comment (jonahkaye) [inline packages/fhir-converter/src/templates/cda/Resources/MedicationStatement.hbs:23]: Fix for this error:
Unknown element 'routeCode' found while parsing

**Review by Goncharo — APPROVED** [review]
:shipit:

**Pull request merged**